optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/valuerange/buildin/temporal/TemporalValueRange.java

```diff
@@ -45,35 +63,43 @@ public TemporalValueRange(Temporal_ from, Temporal_ to, long incrementUnitAmount
         if (!from.isSupported(incrementUnitType) || !to.isSupported(incrementUnitType)) {
             throw new IllegalArgumentException("The " + getClass().getSimpleName()
                     + " must have a incrementUnitType (" + incrementUnitType
-                    + ") that is supported by its from (" + from + ") and to (" + to + ").");
+                    + ") that is supported by its from (" + from.getClass().getSimpleName()
+                    + ") and to (" + to.getClass().getSimpleName() + ").");
         }
         // We cannot use Temporal.until() to check bounds due to rounding errors
         if (from.compareTo(to) > 0) {
             throw new IllegalArgumentException("The " + getClass().getSimpleName()
                     + " cannot have a from (" + from + ") which is strictly higher than its to (" + to + ").");
         }
         long space = from.until(to, incrementUnitType);
-        if (to.equals(from.plus(space + 1, incrementUnitType))) {
+        if (!to.equals(from.plus(space, incrementUnitType))) {
             // Temporal.until() rounds down, but it needs to round up, to be consistent with Temporal.plus()
             space++;
         }
 
         // Fail fast if there's a remainder on amount (to be consistent with other value ranges)
         if (space % incrementUnitAmount > 0) {
             throw new IllegalArgumentException("The " + getClass().getSimpleName()
-                    + " 's incrementUnitAmount (" + incrementUnitAmount
+                    + "'s incrementUnitAmount (" + incrementUnitAmount
                     + ") must fit an integer number of times in the space (" + space
                     + ") between from (" + from + ") and to (" + to + ").");
         }
         // Fail fast if there's a remainder on type (to be consistent with other value ranges)
-        if (!to.equals(from.plus(space, incrementUnitType))) {
+        boolean typeRemainder = true;
+        try {
+            typeRemainder = !to.equals(from.plus(space, incrementUnitType));
+        } catch (DateTimeException e) {
+            // When using extreme ranges, like [LocalDate.MIN-LocalData.MAX) the space may exceed maximum Year value
+            space--; // only needed for the exception message below (otherwise DateTimeException will be thrown)
```

Comment:
+1 but I thought this is similar to catching `NumberFormatException` to see if a string contains a number (you can't do the check without try/catch AFAIK). I'll take a look if there's a way to check if `from.plus(space, incrementUnitType)` overflows maximum Year bound.